features/enlinkd/org.opennms.features.enlinkd.topogen/src/main/java/org/opennms/enlinkd/topogen/TopologyPersister.java

```diff
@@ -0,0 +1,252 @@
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.enlinkd.topogen;
+
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.List;
+
+import org.opennms.netmgt.dao.api.IpInterfaceDao;
+import org.opennms.netmgt.dao.api.NodeDao;
+import org.opennms.netmgt.dao.api.OnmsDao;
+import org.opennms.netmgt.dao.api.SnmpInterfaceDao;
+import org.opennms.netmgt.enlinkd.model.CdpElement;
+import org.opennms.netmgt.enlinkd.model.CdpLink;
+import org.opennms.netmgt.enlinkd.model.IsIsElement;
+import org.opennms.netmgt.enlinkd.model.IsIsLink;
+import org.opennms.netmgt.enlinkd.model.LldpElement;
+import org.opennms.netmgt.enlinkd.model.LldpLink;
+import org.opennms.netmgt.enlinkd.model.OspfLink;
+import org.opennms.netmgt.enlinkd.persistence.api.CdpElementDao;
+import org.opennms.netmgt.enlinkd.persistence.api.CdpLinkDao;
+import org.opennms.netmgt.enlinkd.persistence.api.IsIsElementDao;
+import org.opennms.netmgt.enlinkd.persistence.api.IsIsLinkDao;
+import org.opennms.netmgt.enlinkd.persistence.api.LldpElementDao;
+import org.opennms.netmgt.enlinkd.persistence.api.LldpLinkDao;
+import org.opennms.netmgt.enlinkd.persistence.api.OspfLinkDao;
+import org.opennms.netmgt.model.OnmsIpInterface;
+import org.opennms.netmgt.model.OnmsNode;
+import org.opennms.netmgt.model.OnmsSnmpInterface;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TopologyPersister {
+
+    private final static Logger LOG = LoggerFactory.getLogger(TopologyPersister.class);
+
+    private NodeDao nodeDao;
+    private CdpElementDao cdpElementDao;
+    private IsIsElementDao isIsElementDao;
+    private LldpElementDao lldpElementDao;
+    private CdpLinkDao cdpLinkDao;
+    private IsIsLinkDao isIsLinkDao;
+    private LldpLinkDao lldpLinkDao;
+    private OspfLinkDao ospfLinkDao;
+    private IpInterfaceDao ipInterfaceDao;
+    private SnmpInterfaceDao snmpInterfaceDao;
+
+    private TopologyPersister(
+        final NodeDao nodeDao,
+        final CdpElementDao cdpElementDao,
+        final IsIsElementDao isIsElementDao,
+        final LldpElementDao lldpElementDao,
+        final CdpLinkDao cdpLinkDao,
+        final IsIsLinkDao isIsLinkDao,
+        final LldpLinkDao lldpLinkDao,
+        final OspfLinkDao ospfLinkDao,
+        final IpInterfaceDao ipInterfaceDao,
+        final SnmpInterfaceDao snmpInterfaceDao
+    ) {
```

Comment:
This also looks odd to me.
If we are only using the DAOs for persisting, let's add a `save` method to the `GenericPersistenceAccessor` instead, as that the purpose of that interface/class is to provide exactly this generic behaviour.

Reply:
:+1: makes the code much easier, done.